Mathlib/RingTheory/Spectrum/Prime/ChevalleyComplexity.lean

```diff
@@ -0,0 +1,893 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Andrew Yang
+-/
+import Mathlib.Algebra.Order.SuccPred.WithBot
+import Mathlib.Algebra.Polynomial.CoeffMem
+import Mathlib.Data.DFinsupp.WellFounded
+import Mathlib.RingTheory.Spectrum.Prime.ConstructibleSet
+import Mathlib.RingTheory.Spectrum.Prime.Polynomial
+
+/-!
+# Chevalley's theorem with complexity bound
+
+Chevalley's theorem states that if `f : R → S` is a finitely presented ring hom between commutative
+rings, then the image of a constructible set in `Spec S` is a constructible set in `Spec R`.
```

Comment:
Afaict, this file does not actually contain a proof of this exact statement. But it is an easy corollary of the main result at the bottom of the file.
If my understanding is correct, please indicate so in this module doc.